### PR TITLE
[Merged by Bors] - feat: Add `NumberField.is_primitive_element_of_infinitePlace_lt`

### PR DESCRIPTION
**Pull request by xroblot**
Prove the following
```lean
theorem NumberField.is_primitive_element_of_infinitePlace_lt (x : 𝓞 K)
     {w : InfinitePlace K} (h₁ : x ≠ 0) (h₂ : ∀ ⦃w'⦄, w' ≠ w → w' x < 1)
     (h₃ : IsReal w ∨ |(w.embedding x).re| < 1) : ℚ⟮(x:K)⟯ = ⊤ := by
```

If the place `w` is not real, we need the condition `|(w.embedding x).re| < 1` to ensure `x` is not a real number at the place `w`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 1]
```diff
@@ -523,6 +521,45 @@ theorem prod_eq_abs_norm (x : K) :
   · rw [eq_ratCast, Rat.cast_abs, ← Complex.abs_ofReal, Complex.ofReal_rat_cast]
 #align number_field.infinite_place.prod_eq_abs_norm NumberField.InfinitePlace.prod_eq_abs_norm
 
+theorem ge_one_of_lt_one {w : InfinitePlace K} {a : (𝓞 K)} (ha : a ≠ 0)
+    (h : ∀ ⦃z⦄, z ≠ w → z a < 1) :  1 ≤ w a := by
```
Comment (jcommelin) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:525]:
```suggestion
    (h : ∀ ⦃z⦄, z ≠ w → z a < 1) : 1 ≤ w a := by
```

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 2]
```diff
@@ -523,6 +521,45 @@ theorem prod_eq_abs_norm (x : K) :
   · rw [eq_ratCast, Rat.cast_abs, ← Complex.abs_ofReal, Complex.ofReal_rat_cast]
 #align number_field.infinite_place.prod_eq_abs_norm NumberField.InfinitePlace.prod_eq_abs_norm
 
+theorem ge_one_of_lt_one {w : InfinitePlace K} {a : (𝓞 K)} (ha : a ≠ 0)
```
Comment (jcommelin) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:524]:
```suggestion
theorem one_le_of_lt_one {w : InfinitePlace K} {a : (𝓞 K)} (ha : a ≠ 0)
```

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 3]
```diff
@@ -519,6 +517,45 @@ theorem prod_eq_abs_norm (x : K) :
...
+    · exact pow_pos (pos_iff.mpr ((Subalgebra.coe_eq_zero _).not.mpr ha)) _
+    · refine pow_lt_one (apply_nonneg _ _) ?_ (by rw [mult]; split_ifs <;> norm_num)
+      by_cases hz : z = w
+      · rwa [hz]
+      · exact h hz
+  rw [← Algebra.coe_norm_int, ← Int.cast_one, ← Int.cast_abs, Rat.cast_coe_int, Int.cast_le]
+  exact Int.one_le_abs (Algebra.norm_ne_zero_iff.mpr ha)
+
+open scoped IntermediateField in
+theorem _root_.NumberField.is_primitive_element_of_infinitePlace_lt (x : 𝓞 K)
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:535]:
```suggestion
theorem _root_.NumberField.is_primitive_element_of_infinitePlace_lt {x : 𝓞 K}
```

**Review by riccardobrasca — APPROVED** [review]
Can you add the immediate corollary that `Algebra.adjoin ... = \top`? Thanks!

bors d+

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**